cfme/tests/containers/test_provider_openscap_policy_attached.py

```diff
@@ -0,0 +1,101 @@
+    AttributeToVerify('configuration', 'Last scan', dateparser.parse),
+    AttributeToVerify('compliance', 'Status', lambda val: val.lower() != 'never verified'),
+    AttributeToVerify('compliance', 'History', lambda val: val == 'Available')
+)
+
+
+@pytest.fixture(scope='function')
+def delete_all_container_tasks():
+    delete_all_tasks('AllTasks')
+
+
+@pytest.fixture(scope='function')
+def random_image_instance(appliance):
+    collection = appliance.collections.container_images
+    return random.sample(collection.all(), 1).pop()
+
+
+@pytest.yield_fixture(scope='function')
+def assign_openscap_profile(provider):
+    """assign or unassign policys from provider view
+        Args:
+            assign: Wheter to assign or unassign.
+            policy_profile_names: :py:class:`str` with Policy Profile names.
+    """
+    provider.assign_policy_profiles('OpenSCAP profile')
+    yield
+    provider.unassign_policy_profiles('OpenSCAP profile')
+
+
+@pytest.yield_fixture(scope='function')
+def openscap_assigned_rand_image(provider, random_image_instance):
+    """Returns random Container image that have assigned OpenSCAP policy from provider view.
+    teardown remove this assignment from provider view.
+    """
+
+    # unassign OpenSCAP policy from chosen Image
+    random_image_instance.unassign_policy_profiles('OpenSCAP profile')
+    # assign policy from provider view
+    provider.assign_policy_profiles('OpenSCAP profile')
+    yield random_image_instance
+    provider.unassign_policy_profiles('OpenSCAP profile')
+
+
+def get_table_attr(instance, table_name, attr):
+    # Trying to read the table <table_name> attribute <attr>
+    view = refresh_and_navigate(instance, 'Details')
+    table = getattr(view.entities, table_name, None)
+    if table:
+        return table.read().get(attr)
+
+
+@pytest.mark.polarion('10068')
+def test_check_compliance_provider_policy(provider, soft_assert, delete_all_container_tasks,
+                                          openscap_assigned_rand_image):
+
+    # Prform SSA Scan then check compliance with last know configuration
+    openscap_assigned_rand_image.perform_smartstate_analysis(wait_for_finish=True)
+
+    view = navigate_to(openscap_assigned_rand_image, 'Details')
+    for tbl, attr, verifier in TESTED_ATTRIBUTES__openscap:
+
+        table = getattr(view.entities, tbl)
+        table_data = {k.lower(): v for k, v in table.read().items()}
+
+        if not soft_assert(attr.lower() in table_data, '{} table has missing attribute \'{}\''
+                           .format(tbl, attr)):
+            continue
+        provider.refresh_provider_relationships()
+        wait_for_retval = wait_for(lambda: get_table_attr(openscap_assigned_rand_image, tbl, attr),
+                                   message='Trying to get attribute "{}" of table "{}"'.format(
+                                       attr, tbl),
+                                   delay=5, num_sec=120, silent_failure=True)
```

Comment:
**REQURIED**
```python
wait_for_retval = wait_for(
    get_table_attr,
    func_args=[openscap_assigned_rand_image, tbl, attr],
    message='Trying to get attribute "{}" of table "{}"'.format(attr, tbl),
    delay=5,
    num_sec=120,
    silent_failure=True
)
```

Reply:
Update the wait_for_retval , thanks for correction , its much nicer!